npc_engine/services/utils/config.py

```diff
@@ -97,9 +97,11 @@ def validate_hub_model(models_path: str, model_id: str) -> bool:
             model_correct = False
     except ValueError:
         model_correct = False
-
-    if os.path.exists(config_path):
-        os.remove(config_path)
+    try:
+        if os.path.exists(config_path):
+            os.remove(config_path)
+    except UnboundLocalError:
+        pass

```

Comment:
I would at least log something here. Passing errors in `except` is a code smell :smile:

Reply:
Maybe it makes sense to assign `config_path` outside of `try` if you are not sure if lines 91-93 are run.

Reply:
Good catch, I moved it to the previous try/except block